src/aatoolbox/datasources/datasource.py

```diff
@@ -33,7 +35,15 @@ def __init__(
         country_config: CountryConfig,
         datasource_base_dir: str,
         is_public: bool = False,
+        config_attribute_name: str = None,
```

Comment:
Same, quite confused.

Reply:
Ah sorry, yeah I wasn't sure about this whole thing. I added this because `codab` is now optional in the configuration file. I was thinking that, for a few data sources, we need to confirm that the config file contains the section required for the data source. So here I'm trying to move that functionality to the parent class. You just need to pass the `config_attribute_name` parameter, i.e. the name of the attribute in the configuration file. Then you see at the very beginning of the __ini__: 
```python
        if config_attribute_name is not None and not hasattr(
            country_config, config_attribute_name
        ):
            raise AttributeError(
                f"{config_attribute_name} needs to be implemented in the "
                f"config file. See the documentation for more details."
            )
```

Reply:
Aaaah now I get it. I like the idea! But it was confusing to me, so thinking if we can make it more clear. 
Would it be an idea to have it in a separate function? 
So do 
```
if config_attribute_name is not None:
     self._config_attribute_name_validator()
``` 

Also maybe state "needs to be added to the config file" instead of "implemented"
And maybe name it `config_datasource_name` instead of `config_attribute_name`?
I know these are details though, so I think the idea is good

Reply:
So smart! The only disadvantage might be that I could imagine there are optional config vars but that are used across several datasets (say e.g. the region name for all datasets coming from USGS). If in that case you also have datasource specific vars, then it might get problematic. 

But I think at least for now this approach outweights that unlikely scenario, so I like it and would do it the current way.

Reply:
> The only disadvantage might be that I could imagine there are optional config vars but that are used across several datasets (say e.g. the region name for all datasets coming from USGS). 

That's a good point. We could maybe e.g. put the NDVI section under the USGS section in the config (when we get there). But yeah let's see, it can always be removed / changed!